__tests__/app.test.js

```diff
@@ -0,0 +1,56 @@
+const seed = require("../db/seeds/seed");
+const testData = require("../db/data/test-data");
+const db = require("../db/connection");
+const app = require("../app.js");
+const request = require("supertest");
+
+afterAll(() => {
+  return db.end();
+});
+
+beforeEach(() => {
+  return seed(testData);
+});
+
+describe("GET/api/topics", () => {
+  test("200 and responds with an array of topic objects each of which should have the following properties slug and description", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        expect(topics).toBeInstanceOf(Array);
+        expect(topics).toHaveLength(3);
+        topics.forEach((topic) => {
+          expect(topic).toEqual(
+            expect.objectContaining({
+              slug: expect.any(String),
+              description: expect.any(String),
+            })
+          );
+        });
+      });
+  });
+
+  test("200: returns the correct object key - value pairs", () => {
+    return request(app)
+      .get("/api/topics")
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+        expect(topics[0]).toEqual({
+          slug: "mitch",
+          description: "The man, the Mitch, the legend",
+        });
+      });
+  });
+  test("404:responds with error when passed a route that doesnot exist", () => {
+    return request(app)
```

Comment:
This test is not specific to this endpoint and should probably be in it's own describe block